internal/controllers/httpsedge_controller.go

```diff
@@ -189,71 +197,91 @@ func (r *HTTPSEdgeReconciler) reconcileRoutes(ctx context.Context, edge *ingress
 		secretResolver:   secretResolver{r.Client},
 	}
 
+	edgeRoutes := r.NgrokClientset.HTTPSEdgeRoutes()
+
 	// TODO: clean this up. This is way too much nesting
 	for i, routeSpec := range edge.Spec.Routes {
-		backend, err := tunnelGroupReconciler.findOrCreate(ctx, routeSpec.Backend)
-		if err != nil {
-			return err
-		}
 
 		if routeSpec.IPRestriction != nil {
 			if err := routeModuleUpdater.ipPolicyResolver.validateIPPolicyNames(ctx, edge.Namespace, routeSpec.IPRestriction.IPPolicies); err != nil {
 				if apierrors.IsNotFound(err) {
 					r.Recorder.Eventf(edge, v1.EventTypeWarning, "FailedValidate", "Could not validate ip restriction: %v", err)
 					continue
 				}
-
-				return err
+				return errors.NewErrInvalidConfiguration(err.Error())
```

Comment:
Same comment regarding this error; we don't know if the returned error from `validateIPPolicyNames` is a 5xx or network error or what, and we want to retry those.